### ISSUE TITLE
`danger local --base` inconsistency

### ISSUE THREAD

**Issue by adam-moss** [danger/danger-js#496]
Doing some more testing with `--base`, looks like it is being ignored on the call to`localGetDiff` triggered in the subprocess:

```
DEBUG="*" ./node_modules/.bin/danger local --base develop
 danger:process_runner Starting sub-process run with  +0ms
 danger:localGetDiff git diff develop...HEAD +0ms
 danger:localGetDiff git log develop...HEAD --pretty=format:'{ "sha": "%H", "parents": "%p", "author": {"name": "%an", "email": "%ae", "date": "%ai" }, "committer": {"name": "%cn", "email": "%ce", "date": "%ci" }, "message": "+0ms"},'
 danger:process_runner Preparing to run: /Users/adam.moss/.nvm/versions/node/v9.4.0/bin/node,/Users/adam.moss/projects/project_name/node_modules/danger/distribution/commands/danger-runner.js,--base,develop +29ms
 danger:runDangerSubprocess Running subprocess: node - /Users/adam.moss/projects/project_name/node_modules/danger/distribution/commands/danger-runner.js,--base,develop +0ms
2018-01-24T14:00:53.419Z danger:runner Started Danger runner with

2018-01-24T14:00:53.426Z danger:runner Got STDIN for Danger Run

2018-01-24T14:00:53.441Z danger:localGetDiff git diff master...HEAD

2018-01-24T14:00:53.455Z danger:localGetDiff git log master...HEAD --pretty=format:'{ "sha": "%H", "parents": "%p", "author": {"name": "%an", "email": "%ae", "date": "%ai" }, "committer": {"name": "%cn", "email": "%ce", "date": "%ci" }, "message": "%s"},'

2018-01-24T14:00:53.469Z danger:runner Evaluating dangerfile.js

2018-01-24T14:00:53.470Z danger:inline_runner Started parsing Dangerfile:  dangerfile.js

2018-01-24T14:00:53.470Z danger:inline_runner Finished running dangerfile:  dangerfile.js

2018-01-24T14:00:53.471Z danger:inline_runner Scheduler waiting on: 0 tasks

2018-01-24T14:00:53.471Z danger:inline_runner Finished scheduled tasks

2018-01-24T14:00:54.424Z danger:runner Process has finished with 0 undefined, sending the results back to the host process

 danger:runDangerSubprocess Got JSON results from STDOUT +1s
 danger:runDangerSubprocess child process exited with code 0 +5ms
 danger:executor Got Results back, current settings { stdoutOnly: true,
 verbose: undefined,
 jsonOnly: false,
 dangerID: 'default' } +0ms
 danger:executor Writing to STDOUT: { fails: [],
 warnings: [],
 messages: [],
 markdowns: [],
 scheduled: [] } +2ms
Danger: Passed review, received no feedback.
```

**Comment by peterjgrainger** [comment]
Issue is here where base is hardcoded to master 

https://github.com/danger/danger-js/blob/f7a528779623f8e2be74cc4f100c744f0ef2696f/source/runner/jsonToDSL.ts#L18

Will try and find where to change

**Comment by peterjgrainger** [comment]
Not 100% sure but I don't think the danger-runner.ts file should be ran if doing a local run

Here is where the file is imported

https://github.com/danger/danger-js/blob/f7a528779623f8e2be74cc4f100c744f0ef2696f/source/commands/utils/dangerRunToRunnerCLI.ts#L16

**Comment by orta** [comment]
Yep - it will run, that's the issue - the base needs to go from the local process to the separate runner process somehow

**Comment by orta** [comment]
I'd been wondering about expanding the JSON DSL sent to the sub-process to include individual command settings (it may already be doing this TBH) looks like this is probably the case

**Comment by orta** [comment]
I'm mid some pretty [deep work on Peril](https://github.com/danger/peril/issues/189) so don't want to use my next spare time on this until the peril work is ready - so if someone wants to take a look at that PR that's fine with me

**Comment by peterjgrainger** [comment]
Yeah we can look at it.  Might need a bit of validation before hand to make
sure I'm doing the right thing

**Comment by orta** [comment]
You want to put the fill the `cliArgs` in here, in the local somehow

https://github.com/danger/danger-js/blob/f0a7cf9c452cc52755b4a0e04c7b8a89a2539027/source/dsl/DangerDSL.ts#L17-L46

Then use that in the line you mentioned here: https://github.com/danger/danger-js/blob/f7a528779623f8e2be74cc4f100c744f0ef2696f/source/runner/jsonToDSL.ts#L18

**Comment by peterjgrainger** [comment]
Ok.  I think I get it

**Comment by peterjgrainger** [comment]
I'm trying to add tests for the danger-runner file and saw your note at the top to run this command instead, as pretty fairly, it's hard to unit test a CLI call.

`// yarn build; cat source/_tests/fixtures/danger-js-pr-395.json | env DANGER_FAKE_CI="YEP" DANGER_TEST_REPO='danger/danger-js' DANGER_TEST_PR='395' node distribution/commands/danger-runner.js --text-only
`
I'm having some issues running that command, I get 401 bad credentials response.  I'm guessing because I don't have access to the danger repo.

Was just wondering if you mind me refactoring the non cli parts out into a separate file so I can test it a bit easier?

**Comment by orta** [comment]
Sure, - you probably don't have a `DANGER_GITHUB_API_TOKEN` set up, so your requests are going through un-auth'd

**Comment by peterjgrainger** [comment]
ah, that's probably it.  I've made some unit tests and split it up a little bit.

I'm attaching the CLI args at the point the JSON is parsed.

not sure if you use it but the jest extension for vscode is really good :)  Good choice on using jest, I really like it :)

**Comment by orta** [comment]
Thank, I made that extension ;)
